src/Tring/OutputPrinter.cs

```diff
@@ -16,78 +16,95 @@
 //along with Tring.If not, see<https://www.gnu.org/licenses/>.
 
 using System;
+using System.Threading;
 
 namespace Tring
 {
     internal class OutputPrinter
     {
         private const string timeFormat = "HH:mm:ss";
+        private object _lockObject;
+        private int _lenghtIP, _lenghtPort;
+        private const int _minimumLengthIP = 15;
+        private const int _minimumLengthPort = 4;
 
-        public static void PrintLogEntry(DateTimeOffset startTime, ConnectionResult status)
+        public OutputPrinter(int lenghtIP, int lenghtPort)
         {
-            PrintTime(startTime, status.TimeStamp);
-            PrintRequest(status);
-            PrintResult(status.Connect, status.ConnectionTimeMs);
-            PrintPing(status.PingResult, status.PingTimeMs);
-            PrintLocalInterface(status.LocalInterface);
-            PrintProtocol(status.Request.Port);
-            PrintHostName(status.Request.Url);
+            _lockObject = new object();
+            _lenghtIP = lenghtIP >= _minimumLengthIP ? lenghtIP : _minimumLengthIP;
+            _lenghtPort = lenghtPort >= _minimumLengthPort ? lenghtPort : _minimumLengthPort;
         }
 
-        public static void PrintTable()
+        public void PrintLogEntry(DateTimeOffset startTime, ConnectionResult status, int index)
         {
-            Console.WriteLine("| Time              | IP              | Port  | Connect | Ping    | Local Interface | Protocol | Hostname  ");
-            // example output  | 20:22:22-20:23:33 | 100.100.203.104 | 80222 | Timeout | 1000 ms | 111.111.111.111 | https    | google.com
+            lock (_lockObject)
+            {
+                SetPrintLine(index);
+                PrintTime(startTime, status.TimeStamp);
+                PrintRequest(status);
+                PrintResult(status.Connect, status.ConnectionTimeMs);
+                PrintPing(status.PingResult, status.PingTimeMs);
+                PrintLocalInterface(status.LocalInterface);
+                PrintProtocol(status.Request.Port);
+                PrintHostName(status.Request.Url); 
+            }
+        }
+
+        public void PrintTable()
+        {
+            Console.WriteLine(" | Time              | "+ "IP".PadRight(_lenghtIP)+ " | "+"Port".PadRight(_lenghtPort)+" | Connect | Ping    | "+"Egress".PadRight(_lenghtIP)+" | Protocol | Hostname");
+            // example output   | 20:22:22-20:23:33 | 100.100.203.104 | 80222 | Timeout | 1000 ms | 111.111.111.111 | https    | google.com
         }
 
-        public static void ResetPrintLine()
+        public static void SetPrintLine(int lines = 0)
```

Comment:
Probably a parameter name like `lineNumber` is  a bit more clear than `lines` maybe otherwise `line` in its singular form.